### ISSUE TITLE
Feature Request - Username/Email Login Account Conversion

### ISSUE THREAD

**Issue by MatinatorX** [Ylianst/MeshCentral#1189]
Some of our users connect to MeshCentral infrequently enough that they often forget what username they chose, so I have set "UserNameIsEmail" to make things easier. This is working great for new accounts, however the existing accounts still usernames attached to them. We are not able to log in using the emails attached to these accounts and still have to use the usernames. Would it be possible to add a way to convert these accounts to use email addresses for login instead?

**Comment by Ylianst** [comment]
Good request.

**Comment by Ylianst** [comment]
Just to confirm, existing accounts should still be able to login by entering their existing username in the login screen under "email". Can you confirm this works?

**Comment by MatinatorX** [comment]
Can confirm the older accounts, including my own admin account, can still log in using their username.

**Comment by Ylianst** [comment]
Ok, that is working as expected then. The problem with the conversion is that existing accounts would have their primary keys in the database change which is somewhere risky and has impacts all over the database. This is also why account rename is not practical.

I may still do it, but not a priority and not something that should be done regularly.

**Comment by MatinatorX** [comment]
Understood. I may just have the users create "new" accounts then using their emails and transfer over their permissions manually. the only account I wouldn't be able to do this for would be the admin account but that's no big deal.

**Comment by Ylianst** [comment]
Right. If you have lots of accounts, that will be a problem. If it's just a few, probably just do it manually. The work required to write and test conversion code will be substantial.
